mealy/error_visualizer.py

```diff
@@ -213,41 +233,39 @@ def plot_feature_distributions_on_leaves(self, leaf_selector='all_errors',
         correct_class_idx = 1 - error_class_idx
 
         ranked_feature_ids = rank_features_by_error_correlation(self._error_clf.feature_importances_)
-        if self.pipeline_preprocessor is None:
+        if self._error_analyzer.pipeline_preprocessor is None:
             if top_k_features > 0:
                 ranked_feature_ids = ranked_feature_ids[:top_k_features]
 
-            x, y = self._error_train_x[:, ranked_feature_ids], self._error_train_y
+            x, y = self._error_analyzer._error_train_x[:, ranked_feature_ids], self._error_analyzer._error_train_y
             min_values, max_values = x.min(axis=0), x.max(axis=0)
-            feature_names = self.mpp_feature_names
+            feature_names = self._error_analyzer.preprocessed_feature_names
         else:
-            ranked_feature_ids = [self.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
-                                  ranked_feature_ids]
+            ranked_feature_ids = list(set(self.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
```

Comment:
This attribute does not exist anymore and should be:

```suggestion
            ranked_feature_ids = list(set(self._error_analyzer.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
```